### PR TITLE
Improve cen_instances use common method.

### PR DESCRIPTION
**Pull request by Lexsss**

### CODE REVIEW

File: alicloud/data_source_alicloud_cen_instances.go [hunk 1]
```diff
@@ -112,18 +116,19 @@ func getCenInstances(filters []cbn.DescribeCensFilter, d *schema.ResourceData, m
...
 		if err != nil {
-			return allCens, err
+			return allCens, WrapErrorf(err, DataDefaultErrorMsg, "alicloud_cen_instances", request.GetActionName(), AlibabaCloudSdkGoERROR)
 		}
-		resp, _ := raw.(*cbn.DescribeCensResponse)
+		addDebug(request.GetActionName(), raw)
 
-		if resp == nil || len(resp.Cens.Cen) < 1 {
+		response, _ := raw.(*cbn.DescribeCensResponse)
+		if response == nil || len(response.Cens.Cen) < 1 {
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_cen_instances.go]: response == nil Can be removed

File: alicloud/data_source_alicloud_cen_instances.go [hunk 2]
```diff
@@ -222,35 +235,37 @@ func censDescribeCenAttachedChildInstances(d *schema.ResourceData, cenId string,
...
 		if err != nil {
-			return nil, fmt.Errorf("DescribeCenAttachedChildInstances got an error: %#v.", err)
+			return nil, WrapErrorf(err, DataDefaultErrorMsg, "alicloud_cen_instances", request.GetActionName(), AlibabaCloudSdkGoERROR)
 		}
-		resp, _ := raw.(*cbn.DescribeCenAttachedChildInstancesResponse)
-		if resp != nil && len(resp.ChildInstances.ChildInstance) > 0 {
+		addDebug(request.GetActionName(), raw)
+
+		response, _ := raw.(*cbn.DescribeCenAttachedChildInstancesResponse)
+		if response != nil && len(response.ChildInstances.ChildInstance) > 0 {
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_cen_instances.go]: still

File: alicloud/data_source_alicloud_cen_instances_test.go [hunk 3]
```diff
@@ -1,142 +1,128 @@
...
 
-const testAccCheckAlicloudCenInstancesDataSourceMultiCenIdsConfig = `
-resource "alicloud_cen_instance" "tf-testAccCen" {
-	name = "tf-testAccCenInstancesDataSourceMultiCenIdsConfig"
-	description = "tf-testAccCenConfigDescription"
-	count = 5
+var existCenInstancesMapFunc = func(rand int) map[string]string {
+	return map[string]string{
+		"ids.#":                   "1",
+		"instances.#":             "1",
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_cen_instances_test.go]: names.#

File: alicloud/data_source_alicloud_cen_instances_test.go [hunk 4]
```diff
@@ -1,142 +1,128 @@
...
+		"instances.0.description": "tf-testAccCenConfigDescription",
+	}
 }
 
-data "alicloud_cen_instances" "tf-testAccCen" {
-	ids = ["${alicloud_cen_instance.tf-testAccCen.*.id}"]
+var fakeCenInstancesMapFunc = func(rand int) map[string]string {
+	return map[string]string{
+		"ids.#":       "0",
+		"instances.#": "0",
```
Comment (YY-tomorrow) [inline alicloud/data_source_alicloud_cen_instances_test.go]: still

File: alicloud/resource_alicloud_cen_instance.go [hunk 5]
```diff
@@ -29,11 +29,11 @@ func resourceAlicloudCenInstance() *schema.Resource {
 				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 					value := v.(string)
 					if len(value) < 2 || len(value) > 128 {
-						errors = append(errors, fmt.Errorf("%s cannot be shorter than 2 characters or longer than 128 characters", k))
+						errors = append(errors, WrapError(fmt.Errorf("%s cannot be shorter than 2 characters or longer than 128 characters", k)))
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_cen_instance.go]: fmt.Errorf changed Error

File: alicloud/resource_alicloud_cen_instance.go [hunk 6]
```diff
@@ -69,17 +69,15 @@ func resourceAlicloudCenInstanceCreate(d *schema.ResourceData, meta interface{})
 
 	var cen *cbn.CreateCenResponse
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_cen_instance.go]: cen changed response

File: alicloud/service_alicloud_cen.go [hunk 7]
```diff
@@ -36,18 +36,19 @@ func (s *CenService) DescribeCenInstance(cenId string) (c cbn.Cen, err error) {
 		raw, err := s.client.WithCenClient(func(cbnClient *cbn.Client) (interface{}, error) {
 			return cbnClient.DescribeCens(request)
 		})
-		resp, _ := raw.(*cbn.DescribeCensResponse)
 		if err != nil {
 			if IsExceptedError(err, ParameterCenInstanceIdNotExist) {
-				return GetNotFoundErrorFromString(GetNotFoundMessage("CEN Instance", cenId))
+				return WrapErrorf(err, NotFoundMsg, AlibabaCloudSdkGoERROR)
 			}
 			return err
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_cen.go]: WrapErrorf()

File: alicloud/service_alicloud_cen.go [hunk 8]
```diff
@@ -36,18 +36,19 @@ func (s *CenService) DescribeCenInstance(cenId string) (c cbn.Cen, err error) {
...
 		}
 		addDebug(request.GetActionName(), raw)
-		if len(resp.Cens.Cen) <= 0 || resp.Cens.Cen[0].CenId != cenId {
-			return GetNotFoundErrorFromString(GetNotFoundMessage("CEN Instance", cenId))
+
+		response, _ := raw.(*cbn.DescribeCensResponse)
+		if len(response.Cens.Cen) <= 0 || response.Cens.Cen[0].CenId != cenId {
+			return WrapErrorf(err, NotFoundMsg, ProviderERROR)
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_cen.go]: WrapErrorf(Error(GetNotFoundMessage("CenInstance", id)), NotFoundMsg, ProviderERROR)

File: website/docs/d/cen_instances.html.markdown [hunk 9]
```diff
@@ -35,6 +35,7 @@ The following arguments are supported:
 
 The following attributes are exported in addition to the arguments listed above:
 
+* `names` - A list of CEN instances names.
```
Comment (YY-tomorrow) [inline website/docs/d/cen_instances.html.markdown]: ids

Comment (Lexsss) [reply website/docs/d/cen_instances.html.markdown]: Ids is a Argument, should I write here?

File: alicloud/resource_alicloud_cen_instance.go [hunk 10]
```diff
@@ -67,27 +67,25 @@ func resourceAlicloudCenInstanceCreate(d *schema.ResourceData, meta interface{})
 	request.Description = d.Get("description").(string)
 	request.ClientToken = buildClientToken(request.GetActionName())
 
-	var cen *cbn.CreateCenResponse
+	var response *cbn.CreateCenResponse
 	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
-		req := *request
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cen_instance.go]: Why remove the line 72?

File: alicloud/resource_alicloud_cen_instance_test.go [hunk 11]
```diff
@@ -32,36 +33,36 @@ func testSweepCenInstances(region string) error {
...
-			return cenClient.DescribeCens(req)
+			return cenClient.DescribeCens(describeCensRequest)
 		})
 		if err != nil {
 			return fmt.Errorf("Error retrieving CEN Instances: %s", err)
 		}
-		resp, _ := raw.(*cbn.DescribeCensResponse)
-		if resp == nil || len(resp.Cens.Cen) < 1 {
+		describeCensResponse, _ := raw.(*cbn.DescribeCensResponse)
+		if describeCensResponse == nil || len(describeCensResponse.Cens.Cen) < 1 {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cen_instance_test.go]: Please remove `describeCensResponse == nil`

File: alicloud/resource_alicloud_cen_instance.go (hunk 10 above)
Comment (Lexsss) [reply alicloud/resource_alicloud_cen_instance.go]: Maybe we don't need second level access address.

**Pull request merged**